PlatformIO/src/StateMachine.hpp

```diff
@@ -68,14 +68,27 @@ class Idle : public StateMachine
     if (xSemaphoreTake(wbSemaphore, (TickType_t)10000) == pdTRUE) {
       device->updateColors(COLORS_IDLE);
     }
-    xEventGroupSetBits(audioGroup, STREAM);
-    xSemaphoreGive(wbSemaphore);
-    initHeader(device->readSize, device->width, device->rate);
+    // start streaming audio to the remote endpoint for hotword detection
+    if (config.hotword_detection == HW_REMOTE)
+    {
+      xEventGroupSetBits(audioGroup, STREAM);
+      xSemaphoreGive(wbSemaphore);
+      initHeader(device->readSize, device->width, device->rate);
+    }
   }
 
   void run(void) override {
     if (device->isHotwordDetected() && !hotwordDetected) {
       hotwordDetected = true;
+
```

Comment:
This not needed, the device starts a new session and then enters HotwordDetected state which starts the stream.
It should not be started here.